### PR TITLE
Deepak2

### PR DESCRIPTION
**Pull request by sdeepaknarayanan**
Changed IDW to work properly with other methods. Changed Usage to test on pointwise and gridded interpolation. @apoorvagnihotri merge after having a look

### CODE REVIEW

**Comment by apoorvagnihotri** [comment]
@sdeepaknarayanan Let's discuss #20 first, then I will merge your pull request. Thanks for the pull request.

**Comment by sdeepaknarayanan** [comment]
@apoorvagnihotri ?

File: polire/idw/idw.py [hunk 1]
```diff
@@ -95,13 +105,17 @@ def _predict_grid(self, x1lim, x2lim):
...
         new_grid = np.zeros((len(xx),len(yy)))
-
         for i in range(len(new)):
             # Storing the source data point values into the corresponding grid locations.
             # As from above, we store all the values needed in the new_list. 
             x = new[i][1][0]
             y = new[i][1][1]
-            new_grid[x][y] = new[i][0][2]
+            if new[i][0][2] == 0.0:
+                new_grid[x][y] = 1e-20
```
Comment (apoorvagnihotri) [inline polire/idw/idw.py]: #7 ? Can we not avoid this?

Comment (sdeepaknarayanan) [reply polire/idw/idw.py]: new is a list. I can (of course) make it more efficient. I'll update this in the next PR

**Pull request merged**